src/webgl/p5.Matrix.js

```diff
@@ -1,717 +1,1324 @@
+    z = mB[10];
+    w = mB[11];
+    mC[8] = x * a0 + y * a4 + z * a8 + w * a12;
+    mC[9] = x * a1 + y * a5 + z * a9 + w * a13;
+    mC[10] = x * a2 + y * a6 + z * a10 + w * a14;
+    mC[11] = x * a3 + y * a7 + z * a11 + w * a15;
+
+    x = mB[12];
+    y = mB[13];
+    z = mB[14];
+    w = mB[15];
+    mC[12] = x * a0 + y * a4 + z * a8 + w * a12;
+    mC[13] = x * a1 + y * a5 + z * a9 + w * a13;
+    mC[14] = x * a2 + y * a6 + z * a10 + w * a14;
+    mC[15] = x * a3 + y * a7 + z * a11 + w * a15;
+  } else {
+    a0 = mA[0];
+    a1 = mA[1];
+    a2 = mA[2];
+    a3 = mA[3];
+    a4 = mA[4];
+    a5 = mA[5];
+    a6 = mA[6];
+    a7 = mA[7];
+    a8 = mA[8];
+
+    x = mB[0];
+    y = mB[1];
+    z = mB[2];
+    mC[0] = x * a0 + y * a3 + z * a6;
+    mC[1] = x * a1 + y * a4 + z * a7;
+    mC[2] = x * a2 + y * a5 + z * a8;
+
+    x = mB[3];
+    y = mB[4];
+    z = mB[5];
+    mC[3] = x * a0 + y * a3 + z * a6;
+    mC[4] = x * a1 + y * a4 + z * a7;
+    mC[5] = x * a2 + y * a5 + z * a8;
+
+    x = mB[6];
+    y = mB[7];
+    z = mB[8];
+    mC[6] = x * a0 + y * a3 + z * a6;
+    mC[7] = x * a1 + y * a4 + z * a7;
+    mC[8] = x * a2 + y * a5 + z * a8;
   }
-  det = 1.0 / det;
-  this.mat3[0] = b01 * det;
-  this.mat3[1] = (-a22 * a01 + a02 * a21) * det;
-  this.mat3[2] = (a12 * a01 - a02 * a11) * det;
-  this.mat3[3] = b11 * det;
-  this.mat3[4] = (a22 * a00 - a02 * a20) * det;
-  this.mat3[5] = (-a12 * a00 + a02 * a10) * det;
-  this.mat3[6] = b21 * det;
-  this.mat3[7] = (-a21 * a00 + a01 * a20) * det;
-  this.mat3[8] = (a11 * a00 - a01 * a10) * det;
+
   return this;
 };
 
 /**
- * transposes a 3x3 p5.Matrix by a mat3
- * @method transpose3x3
- * @param  {Number[]} mat3 1-dimensional array
+ * Matrix Pre-Multiplication: mC = mA * mB
+ *
+ * operation: mC(mdst|this) = mA(mleft) * mB(this)
+ *
+ * note: both matrices must be of the same size ... mat4 or mat3
+ *
+ * @method mult
+ * @param {p5.Matrix | Float32Array | Array} mleft matrix (left side)
+ * @returns {p5.Matrix} this matrix, as the result of the multiplication
  * @chainable
  */
-p5.Matrix.prototype.transpose3x3 = function(mat3) {
-  var a01 = mat3[1],
-    a02 = mat3[2],
-    a12 = mat3[5];
-  this.mat3[1] = mat3[3];
-  this.mat3[2] = mat3[6];
-  this.mat3[3] = a01;
-  this.mat3[5] = mat3[7];
-  this.mat3[6] = a02;
-  this.mat3[7] = a12;
+p5.Matrix.prototype.premult = function(mleft) {
+  mleft.mult(this, this);
   return this;
 };
 
 /**
- * converts a 4x4 matrix to its 3x3 inverse tranform
- * commonly used in MVMatrix to NMatrix conversions.
- * @method invertTranspose
- * @param  {p5.Matrix} mat4 the matrix to be based on to invert
+ * If this matrix owns a p5-reference it will respect the p5._angleMode flag.
+ *
+ * TODO: find a better way for doing this
+ *       or, get rid of anglemode and the p5 instance
+ *
+ * @method assertAngleMode
+ * @param {Number} angle
+ * @returns {Number} the angle (deg or rad)
+ */
+p5.Matrix.prototype.assertAngleMode = function(angle) {
```

Comment:
I think this has been superseded by new changes. I can address this when I handle merge conflicts (I spent some time on it while testing this afternoon).